src/useCases/__tests__/processRequest.test.ts

```diff
@@ -6,15 +6,25 @@ describe('processRequest', () => {
   it('should return mock response', async () => {
     const { env, envData } = makeFakeCapabilities();
 
-    const actual = await processRequest(envData.mock.aMockRequest)(env)();
+    const actual = await processRequest(envData.mock.anHttpRequest)(env)();
     const expected = envData.requestResponse.aRequestResponse.response;
     expect(actual).toStrictEqual(E.right(expected));
   });
   it('should record the request-response pair', async () => {
     const { env, envData } = makeFakeCapabilities();
 
-    await processRequest(envData.mock.aMockRequest)(env)();
+    await processRequest(envData.mock.anHttpRequest)(env)();
     const expected = envData.requestResponse.aRequestResponse;
     expect(env.requestResponseWriter.record).nthCalledWith(1, expected);
   });
+  it('should return the response defined by user if any', async () => {
+    const { env, envData } = makeFakeCapabilities();
+
+    const actual = await processRequest(
+      envData.customResponseDefinition.anHttpRequest
+    )(env)();
```

Comment:
```suggestion
    const actual = await processRequest(
      {...envData.customResponseDefinition.anHttpRequest}
    )(env)();
```
Here I would make a copy of the object I want to pass; if you do this, you will see the test failing.
As you can see in this [example that uses the `===` operator,](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality#comparing_objects), the comparison of two different objects with same properties and values is `false`.
You can find more details about how the strict equal operator works in the [description section](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality#description).